zebra-consensus/src/transaction/tests.rs

```diff
@@ -86,8 +86,34 @@ fn fake_v5_transaction_with_orchard_actions_has_inputs_and_outputs() {
     // guaranteed structurally by `orchard::ShieldedData`)
     insert_fake_orchard_shielded_data(&mut transaction);
 
-    // If a transaction has at least one Orchard shielded action, it should be considered to have
-    // inputs and/or outputs
+    // The check will fail if it haves no flags
+    assert_eq!(
+        check::has_inputs_and_outputs(&transaction),
+        Err(TransactionError::NoInputs)
+    );
+
+    // If we add ENABLE_SPENDS flag it will pass the inputs check but fails with the outputs
+    let shielded_data = insert_fake_orchard_shielded_data(&mut transaction);
```

Comment:
I think there should be no need to get a new transaction with a new call to `insert_fake_orchard_shielded_data()` each time we want to test something however i had troubles with the mutable borrows and passing them multiple times.

Reply:
Maybe @jvff can help here?

Reply:
I created a low priority ticket for this (https://github.com/ZcashFoundation/zebra/issues/2409) and i added a TODO comment into the code for it in https://github.com/ZcashFoundation/zebra/pull/2398/commits/05019a9d4d5b7a18517489029f9a9a19171b76c4